### PR TITLE
"vines", "plants" faction gives immunity to kuzdu

### LINKED ISSUE #18509: Kuzdu/space vines still attack podpeople
RIP a certain botanist.

### PR DESCRIPTION
**Pull request by coiax**
Adding "vines" and "plants" to podpeople's factions only makes them
immune to the snare of the venus human trap. Now added some checks to
spacevine.dm which checks for vine immunity, and a check in
mob/living.dm (apparently).

:cl: coiax
rscadd: Podpeople (and venus human traps) are now no longer damaged,
poisoned or obstructed by space vines. Explosive vines still damage
them, because there's an actual explosion.
/:cl:

Fixes #18509
- Simple animals now factor in bomb armor when subject to explosions.
- Space covering vines will no longer replace transit space turfs.

### CODE REVIEW

**Comment by KorPhaeron** [comment]
when do the daisy buffs stop

File: code/modules/events/spacevine.dm [hunk 1]
```diff
@@ -571,3 +573,16 @@
 		override += SM.process_temperature(src, temp, volume)
 	if(!override)
 		qdel(src)
+
+/obj/effect/spacevine/CanPass(atom/moveable/mover, turf/target, height=0)
```
Comment (Jordie0608) [inline code/modules/events/spacevine.dm]: atom/**movable**/mover

**Comment by Core0verload** [comment]
inb4 seed vault podmen kill all the bosses with space kudzu

**Comment by Xhuis** [comment]
What's next, growable bombs?

**Comment by Kromgar** [comment]
I'm sure it is Xhuis what kind of monster would make Growable bombs that are weaker than Syndibombs?

**Comment by KorPhaeron** [comment]
In all seriousness I know that space vine immunity is fun and fluffy for plant men but the vines are already round enders without the person making them also being immune/hiding in them indefinitely

**Comment by coiax** [comment]
Making podpeople immune to kuzdu was the point of #18424 which is already merged. This just makes kudzu actually respect those factions.

**Comment by MMMiracles** [comment]
Science was nothing more than a distraction for the true department powerhouse: Botany.

File: code/modules/mob/living/living.dm [hunk 2]
```diff
@@ -225,7 +225,9 @@ Sorry Giacom. Please don't be mad :(
 /mob/living/proc/InCritical()
 	return (src.health < 0 && src.health > -95 && stat == UNCONSCIOUS)
 
-/mob/living/ex_act(severity, target)
+/mob/living/ex_act(severity, origin)
+	if(istype(origin, /datum/spacevine_mutation) && isvineimmune(src))
+		return
```
Comment (phil235) [inline code/modules/mob/living/living.dm]: snowflake :-1:

**Comment by ChangelingRain** [comment]
Vines are already bad enough with the whole 'you can't click under them except with altclick' shit

File: code/modules/mob/living/living.dm (hunk 2 above)
Comment (Razharas) [inline code/modules/mob/living/living.dm]: Second argument for the ex_act was added specifically to have special cases like this and c4 ones

**Comment by coiax** [comment]
That's a design choice in the layering/opacity of the vines; we can change that.

**Comment by kevinz000** [comment]
I love how botany keeps getting new """features""" and buffs while not getting a single nerf/rebalance.
And by rebalance I don't mean making kudzu welderproof too.

**Comment by KorPhaeron** [comment]
That last pr was probably a bad idea as well then. This would be pretty awful to deal with.

**Comment by KorPhaeron** [comment]
Maybe if you could click through vines but I am not sure it's worth it for this

**Comment by coiax** [comment]
I literally just made this because I saw someone get murderised as a podperson by vines thinking that they were immune because there was previously a PR that said it did that.

**Comment by Zilenan91** [comment]
What the fuck Kudzu is welderproof now? That needs to be reverted because that was one of the only things that made it not ridiculous

**Comment by coiax** [comment]
No, fireproof kudzdu is welder proof. That was a bug fix.

What I think needs to be done is move kuzdu from a probability based chance of removal to flat HP.

**Comment by kevinz000** [comment]
So basiclly fireproof hardened is almost invulnerable now
And aggressive flowering instakills anyone touching or crossing them by calling ex act on them
Which by the way can delimb
And mechs are the only counter as you can move through it with mechs.

**Comment by optimumtact** [comment]
You're supposed to use weedkiller crates

File: code/modules/mob/living/living.dm (hunk 2 above)
Comment (optimumtact) [inline code/modules/mob/living/living.dm]: I think that's messy though, surely there is a better way, add an explosion_type argument to the ex_act?

**Comment by Zilenan91** [comment]
The weedkiller crates you can't get to because hardened vines stop movement and spread faster than plasma in the air

**Comment by Shadowlight213** [comment]
@kevinz000 oh yes. We had to deal with those on pre hub bagil lowpop a lot. Once they mutated fireproof and hardened it was basically call the shuttle we're fucked. We sometimes managed to fight them off with weedkillers.
Man up and deal with it. If 10 pop managed to deal with it, surely high pop can.

**Comment by coiax** [comment]
As a quick aside, podpeople are specifically hyper vulnerable to weedkiller.
10u of plantbgone will deal 75 toxin damage to a podperson, and a weedkiller grenade (with full exposure) will add at least double that to a podperson.

File: code/modules/mob/living/living.dm (hunk 2 above)
Comment (Razharas) [inline code/modules/mob/living/living.dm]: Target is pretty much data argument where you can put anything and let other thigns try and react to this things accordingly
So its just ununifies system of explosion types

**Comment by optimumtact** [comment]
So maybe if we put weedkiller crates in the emergency maint closets?

**Comment by Zilenan91** [comment]
That'd work

**Comment by optimumtact** [comment]
This needs changes to make it palatable to the design lead or it isn't going anywhere.

**Comment by KorPhaeron** [comment]
I know this is a bugfix but the original change was a bad idea

**Comment by kevinz000** [comment]
now that I got kudzu nerfed into th- I mean, bugfixed, this isn't that bad of a bad idea as the crew can actually kill the weeds instead of getting instantly wrecked.

**Comment by coiax** [comment]
The only problems with this PR are the design issues with the second argument to ex_act, which is currently ONLY used in c4 elsewhere in the codebase. If there's a better way of making vine people immune to the ex_act of aggressive vines, I'm all ears.

**Comment by KorPhaeron** [comment]
They could just suck it up and stay away from explosives

**Comment by KorPhaeron** [comment]
Or the vine could be hacky and check for a mob with the faction before trying to bomb a turf

File: code/modules/mob/living/carbon/human/human.dm [hunk 3]
```diff
@@ -120,6 +120,8 @@
 	var/b_loss = null
 	var/f_loss = null
 	var/bomb_armor = getarmor(null, "bomb")
+	if(istype(ex_target, /datum/spacevine_mutation) && isvineimmune(src))
```
Comment (optimumtact) [inline code/modules/mob/living/carbon/human/human.dm]: man this is wack, but it's not really your fault

**Comment by optimumtact** [comment]
I think it's fine, but if someone could cast their eye over the ex_act and look at a potential refactor to improve that would be :star:
